craft_parts/parts.py

```diff
@@ -1073,6 +1097,20 @@ def part_dependencies(
     return dependencies
 
 
+def part_dependencies_overlay_bootstrap(
```

Comment:
Would `dependencies_that_bootstrap_overlay` be a clearer name?

Reply:
Yes. I struggled finding proper names for some methods because they are doing very specific things.